### PR TITLE
Move and improve tax integration specs, include shipping

### PR DESCRIPTION
**Pull request by mamhoff**
This moves the taxation integration specs into their own file. 

I'm also using the new address factory functionality to actually test for addresses being taxed correctly rather than zones being taxed correctly in order to reduce mingling specs with implementation.

### CODE REVIEW

**Comment by mamhoff** [comment]
Now ready to merge. This unearthes a whole bunch of weirdo things about shipping rate and shipment tax calculation, many of which will be fixed with #759. Once this is in, I'll rebase that PR so we get a better idea of how it improves things.

File: core/spec/models/spree/tax/taxation_integration_spec.rb [hunk 1]
```diff
@@ -0,0 +1,801 @@
...
+      create :shipping_method,
+             cost: 8.00,
+             shipping_categories: [books_shipping_category],
+             tax_category: books_category,
+             zones: [eu_zone, world_zone]
+    end
+
+    let!(:sweater_shipping_method) do
+      create :shipping_method,
+             cost: 16.00,
```
Comment (tvdeyen) [inline core/spec/models/spree/tax/taxation_integration_spec.rb]: Is this a real world use case? Shouldn't we test different rates for each zone? I.e.: 6€ in German zone, 12€ in EU zone, 20€ for world?

Comment (mamhoff) [inline core/spec/models/spree/tax/taxation_integration_spec.rb]: That would be more realistic, but the system under test here is the
taxation system. I'm assuming the shipping system works - the additional
benefit is that the rates used are always recognizable: 16 for sweaters, 8
for books and 2 for premium downloads ;)
Am 05.02.2016 19:32 schrieb "Thomas von Deyen" <notifications@github.com>:

**Comment by jhawthorn** [comment]
:+1: awesome

**Comment by Senjai** [comment]
:+1:

**Comment by mamhoff** [comment]
Can we merge this soon-ish?

**Pull request merged**